server/routes/oembed/oembed.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Endpoint for oEmbed API, for embedding web components using oEmbed"""
+
+import json
+import math
+import re
+
+from flask import Blueprint
+from flask import request
+from flask import Response
+
+from server import cache
+import server.lib.config as libconfig
+
+DEFAULT_WIDTH = 500
+DEFAULT_HEIGHT = 400
+
+# Define blueprint
+bp = Blueprint("oembed", __name__, url_prefix="/oembed")
+
+# Set allowed url values
+# We want to allow only datacommons urls with /chart endpoints to be passed in
+# Hostname changes based on current config (local vs autopush/prod)
+url_regex = "https?://*\.datacommons.\org/chart*"
+cfg = libconfig.get_config()
+if cfg.LOCAL:
+  url_regex = "http://(127\.0\.0\.1|localhost):8080/chart*"
+
+
+@bp.route("/", strict_slashes=False)
+@cache.cache.cached(timeout=cache.TIMEOUT, query_string=True)
+def render_chart():
+  response_format = request.args.get("format", type=str, default="json")
+  url = request.args.get("url", type=str)
+  if not url or not re.match(url_regex, url):
+    # reject request if url not matching allowed pattern or not provided
+    return "error: must provide a valid url", 400
+  max_width = request.args.get("maxwidth", type=int)
+  max_height = request.args.get("maxheight", type=int)
+
+  width = math.min(max_width, DEFAULT_WIDTH) if max_width else DEFAULT_WIDTH
```

Comment:
nit: could just set the default width and height when getting the arg values in lines 50 and 51? then here, just need to do the math.min() part

Reply:
Thanks! Implemented. Much cleaner now.